src/puppetlabs/puppetdb/command.clj

```diff
@@ -956,6 +964,7 @@
    (finally (when command-shovel (.interrupt command-shovel)))
    (finally (some-> broadcast-threadpool pool/shutdown-unbounded))
    (finally (some-> consumer-threadpool pool/shutdown-gated))
+   (finally (when blocked? (locking blocked? (reset! blocked? false) (.notifyAll blocked?))))
```

Comment:
After a bit more thought (and if I'm thinking straight), I believe we can just remove this, given the `(.interrupt command-shovel)` above.  i.e. the interrupt should cause it to throw back to the catch in the shovel thread and then exit (even) if the thread actually is blocked in `.wait`, and other than the shovel thread itself, there's nothing else to notify.  It also might be better to stay blocked during shutdown anyway, i.e. shutting down perhaps shouldn't (even temporarily) unblock us, from a semantic perspective.

Reply:
OK, so then, if I follow you, perhaps we can remove that `finally` code without any issue.